Facebook.iOS/ShareKit/source/Facebook.ShareKit/ApiDefinition.cs

```diff
@@ -0,0 +1,1310 @@
+
+	// @protocol FBSDKShareOpenGraphValueContaining <NSObject, NSSecureCoding>
+	[Model (AutoGeneratedName = true)]
+	[Protocol]
+	[BaseType (typeof (NSObject), Name = "FBSDKShareOpenGraphValueContaining")]
+	interface ShareOpenGraphValueContaining : INSSecureCoding {
+
+		// @required -(NSArray *)arrayForKey:(NSString *)key;
+		[Abstract]
+		[Export ("arrayForKey:")]
+		NSObject [] ArrayForKey (string key);
+
+		// @required -(void)enumerateKeysAndObjectsUsingBlock:(void (^)(NSString *, id, BOOL *))block;
+		[Abstract]
+		[Export ("enumerateKeysAndObjectsUsingBlock:")]
+		unsafe void EnumerateKeysAndObjectsUsingBlock (ShareOpenGraphValueContainingEnumerateHandle handle);
+
+		// @required -(NSEnumerator *)keyEnumerator;
+		[Abstract]
+		[Export ("keyEnumerator")]
+		NSEnumerator KeyEnumerator ();
+
+		// @required -(NSNumber *)numberForKey:(NSString *)key;
+		[Abstract]
+		[Export ("numberForKey:")]
+		NSNumber NumberForKey (string key);
+
+		// @required -(NSEnumerator *)objectEnumerator;
+		[Abstract]
+		[Export ("objectEnumerator")]
+		NSEnumerator ObjectEnumerator ();
+
+		// @required -(FBSDKShareOpenGraphObject *)objectForKey:(NSString *)key;
+		[Abstract]
+		[Export ("objectForKey:")]
+		ShareOpenGraphObject ObjectForKey (string key);
+
+		// @required -(id)objectForKeyedSubscript:(NSString *)key;
+		[Abstract]
+		[Export ("objectForKeyedSubscript:")]
+		NSObject ObjectForKeyedSubscript (string key);
+
+		// @required -(void)parseProperties:(NSDictionary *)properties;
+		[Abstract]
+		[Export ("parseProperties:")]
+		void ParseProperties (NSDictionary properties);
+
+		// @required -(FBSDKSharePhoto *)photoForKey:(NSString *)key;
+		[Abstract]
+		[Export ("photoForKey:")]
+		SharePhoto PhotoForKey (string key);
+
+		// @required -(void)removeObjectForKey:(NSString *)key;
+		[Abstract]
+		[Export ("removeObjectForKey:")]
+		void RemoveObjectForKey (string key);
+
+		// @required -(void)setArray:(NSArray *)array forKey:(NSString *)key;
+		[Abstract]
+		[Export ("setArray:forKey:")]
+		void SetArray (NSObject [] array, string key);
+
+		// @required -(void)setNumber:(NSNumber *)number forKey:(NSString *)key;
+		[Abstract]
+		[Export ("setNumber:forKey:")]
+		void SetNumber (NSNumber number, string key);
+
+		// @required -(void)setObject:(FBSDKShareOpenGraphObject *)object forKey:(NSString *)key;
+		[Abstract]
+		[Export ("setObject:forKey:")]
+		void SetObject (ShareOpenGraphObject graphObject, string key);
+
+		// @required -(void)setPhoto:(FBSDKSharePhoto *)photo forKey:(NSString *)key;
+		[Abstract]
+		[Export ("setPhoto:forKey:")]
+		void SetPhoto (SharePhoto photo, string key);
+
+		// @required -(void)setString:(NSString *)string forKey:(NSString *)key;
+		[Abstract]
+		[Export ("setString:forKey:")]
+		void SetString (string aString, string key);
+
+		// @required -(void)setURL:(NSURL *)URL forKey:(NSString *)key;
+		[Abstract]
+		[Export ("setURL:forKey:")]
+		void SetUrl (NSUrl Url, string key);
```

Comment:
I think we should wrap `SetURL` to be consistent here.

Reply:
This is a protocol, I think we cannot use [Wrap] here. Should I return the name as before?